docs/src/adr/005adr-json.md

```diff
@@ -95,21 +162,27 @@ The `value` field depends on the kind of `TlaValue`:
 1. For `TlaBool`: a JSON Boolean
 1. For `TlaInt(bigIntValue)`: 
   1. If `bigIntValue.isValidInt`: a JSON number
-  2. Otherwise: `{ bigInt: bigIntValue.toString() }`
+  2. Otherwise: `{ "bigInt: bigIntValue.toString() }`
 1. For `TlaDecimal(decValue)`: a JSON string `decValue.toString`
 
+The reason for the non-uniform treatment of integers, is that Apalache encodes its TLA+ integers as `BigInt`, which means that it permits values, for which `.isValidInt` does not hold. 
+
+While it might seem more natural to encode the entire `TlaValue` as a JSON primitive, without the added object layer, we would have a much tougher time deserializing.
+We would need a) a sensible encoding of `BigInt` values, which are not valid integers, and b) a way to distinguish both variants of `BigInt`, as well as decimals, when deserializing (since JSON is not typed).
+We could encode all values as strings, but they would be similarly indistinguishable when deserializing. Importantly, the `type` field of the `ValEx` expression is not guaranteed to contain a hint, as it could be `Untyped`
```

Comment:
You could cite [RFC7159](https://tools.ietf.org/html/rfc7159#page-6) to justify why we cannot just use unbounded numbers.

Reply:
I don't think it's about the capabilities of JSON numbers, it has more to do with the unnecessary complexity of interpreting those JSON values as the correct Scala type, which we can avoid pretty much entirely by just having a `kind` field hint.

Reply:
The reason for not using JSON number is that they have a bounded range, something like `[-(2**53)+1, (2**53)-1]`, which I am sure a few people know about.